engines/director/score.cpp

```diff
@@ -386,15 +393,20 @@ void Score::update() {
 	}
 
 	if (!_vm->_playbackPaused) {
-		if (_nextFrame)
-			_currentFrame = _nextFrame;
+		if (_nextFrame) {
+			// With the advent of demand loading frames and due to partial updates, we rebuild our channel data
+			// when jumping.
+			if (_nextFrame != _curFrameNumber)
+				rebuildChannelData(_nextFrame);
```

Comment:
The logic here is, that _loadFrame()_ is supposed to load any frame without modifications to channelData, (e we are not modifiying/recalculating channel data for each frames, so when movie is in normal execution, it will go from frame X to frame X+1, in this case rebuilding channel data is not needed and thus use normal loadFrame, however the moment there is jump, now according to these delta updates there could be big issues, like for example our channel data has been updated with regards to frame 10, and there is jump to frame 5, then this backward jump with unchanged _channelData_ will be a cause for misbehaviour (I verified this, it happened).

The fix to this, as in here, is that in normal linear frames loading, we are not rebuilding channelData, however if there is any kind of jump (which is not to itself), then this will recreate all of the channelData and thus prepare for next frame loading.

It can be bit confusing that we are _reading it again_, however in essence we are not wanting to read this frame, we are just reading from frames 1....n and creating proper channelData, so once we read the requested frame using loadFrame, there isn't any redundant data that has been carried forward by jump.